frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -346,6 +354,34 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
     }
   };
 
+  cancelUpload = async () => {
+    const { uploadId, resumableUpload, datasetId, datastoreUrl } = this.state;
+    resumableUpload.pause();
+    const shouldCancel = await confirmAsync({
+      title:
+        "Cancelling the running upload will delete already uploaded files on the server and cannot be undone. Are you sure you want to cancel the upload?",
+      okText: "Yes, Cancel the upload.",
```

Comment:
I’d suggest overwriting cancelText as well (default is cancel, I believe, which is confusing here)
```suggestion
      okText: "Yes, Cancel the upload",
      cancelText: "No, keep it running",
```